### ISSUE TITLE
Categorical seems to be miscalculated

### ISSUE THREAD

**Issue by ChengYen-Tang** [dotnet/TorchSharp#836]
It looks like there is a problem with using parameters(logits).

Categorical: set logits

python:
```
>>> from torch.distributions import Categorical
>>> import torch as th
>>> th.manual_seed(1)
>>> nSamples = int(5e6)
>>> actionLogits = th.rand(nSamples, 2)
distribution = Categorical(logits=actionLogits)
>>> actions = distribution.sample()
>>> entropy = distribution.entropy()
>>> log_prob = distribution.log_prob(actions)
>>> entropy.mean()
tensor(0.6733)
>>> log_prob.mean()
tensor(-0.6733)
```
C#:
```
using TorchSharp;
using TorchSharp.Modules;
using static TorchSharp.torch;

int nSamples = Convert.ToInt32(5e6);
random.manual_seed(1);
Tensor actionLogits = rand(nSamples, 2);
Categorical distribution = new(logits: actionLogits);
Tensor actions = distribution.sample();
Tensor entropy = distribution.entropy();
Tensor log_prob = distribution.log_prob(actions);
Console.WriteLine(entropy.mean().ToDouble());
Console.WriteLine(log_prob.mean().ToDouble());

output:
0.4533173143863678
-0.6858589053153992
```

Categorical: set probs

python:
```
>>> from torch.distributions import Categorical
>>> import torch as th
>>> th.manual_seed(1)
>>> nSamples = int(5e6)
>>> actionLogits = th.rand(nSamples, 2)
distribution = Categorical(probs=actionLogits)
>>> actions = distribution.sample()
>>> entropy = distribution.entropy()
>>> log_prob = distribution.log_prob(actions)
>>> entropy.mean()
tensor(0.5639)
>>> log_prob.mean()
tensor(-0.5642)
```
C#:
```
using TorchSharp;
using TorchSharp.Modules;
using static TorchSharp.torch;

int nSamples = Convert.ToInt32(5e6);
random.manual_seed(1);
Tensor actionLogits = rand(nSamples, 2);
Categorical distribution = new(probs: actionLogits);
Tensor actions = distribution.sample();
Tensor entropy = distribution.entropy();
Tensor log_prob = distribution.log_prob(actions);
Console.WriteLine(entropy.mean().ToDouble());
Console.WriteLine(log_prob.mean().ToDouble());

output:
0.5639296174049377
-0.5641133189201355
```

**Comment by ChengYen-Tang** [comment]
Same problem with MultiCategoricalDistribution

**Comment by NiklasGustafsson** [comment]
@ChengYen-Tang -- I identified the bug. This one will also require a NuGet update. The workaround is the same -- use probs instead of logits when creating the distribution.

I don't know which second distribution you're referring to, there's no `MultiCategorical` distribution.

**Comment by ChengYen-Tang** [comment]
> @ChengYen-Tang -- I identified the bug. This one will also require a NuGet update. The workaround is the same -- use probs instead of logits when creating the distribution.
> 
> I don't know which second distribution you're referring to, there's no `MultiCategorical` distribution. There is `Multinomial,` which is based on Categorical, so it's probably the same bug.

Sorry, this is my custom object, it is based on Categorical🤣
